charms/mlflow-server/src/charm.py

```diff
@@ -331,6 +331,10 @@ def _define_secrets(self, obj_storage, mysql):
             {"name": f"{self.charm_name}-db-secret", "data": _db_secret_dict(mysql=mysql)},
         ]
 
+    def _gen_obj_storage_endpoint_url(self, obj_storage):
+        """Generate object storage endpoint URL."""
+        return f"http://{obj_storage['service']}.{obj_storage['namespace']}:{obj_storage['port']}"
```

Comment:
Should we add a check to ensure  `obj_storage`  actually has those keys you are calling?

Reply:
Older code never had those checks. But I guess it is good idea. Will return empty string then.

Reply:
Done.